src/Corrade/Utility/Test/DirectoryTest.cpp

```diff
@@ -585,12 +587,35 @@ void DirectoryTest::mkpathNoPermission() {
     if(Directory::exists("/nope"))
         CORRADE_SKIP("Can't test because the destination might be writeable");
 
+    /* Ensure errors are printed in English */
+    std::setlocale(LC_ALL, "en_US.utf8");
```

Comment:
```suggestion
    /* Ensure errors are printed in English */
    const char* currentLocale = std::setlocale(LC_ALL, nullptr);
    std::setlocale(LC_ALL, "en_US.utf8");
    ScopeGuard restoreLocale{currentLocale, [](const char* locale) {
        std::setlocale(LC_ALL, locale);
    }};
```

... and put it together with the `mkpath()` in a scope that's as small as possible -- I don't know what all `setlocale()` affects, so better be safe than waste time debugging weird interactions on weird platforms later :)

Reply:
✅